test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -56,18 +56,37 @@ struct collection_t {
 /* Representation of the collections in memory. */
 class database {
     public:
-    const std::vector<std::string>
-    get_collection_names() const
+    std::mutex &
+    get_mtx()
+    {
+        return (_mtx);
+    }
+
+    /*
+     * The database's methods take a `std::lock_guard`. We don't do anything with the guard itself,
+     * but its a way to make sure that callers don't forget to lock and accidentally shoot
+     * themselves in the foot.
+     */
+    std::vector<std::string>
+    get_collection_names(const std::lock_guard<std::mutex> &) const
```

Comment:
It's designed this way to avoid excessive locking and unlocking. If you're doing a bunch of things to it (like the validator is doing), you don't want to be locking for every single operation. Just lock once and do your thing.

Reply:
How does this actually guarantee that the lock is taken? I've not seen this style of locking before.

Reply:
As discussed, we're going to chang the API to lock within each method now.